icsharpcode/CodeConverter
# Issue/774

Fixes #774

### Problem
Methods handling events on a WithEvents property from a base class are not wired up after conversion

### Solution

* Look for methods which handle events on properties not defined within the type.
* Override that property to add the appropriate handlers
* In the base constructor, assign the property so that it runs the overridden code upon construction without further effort

### Trade-offs

* For the sake of a simple output, assume there's no type outside the compilation which inherits from a type and overrides a WithEvents property within it (e.g. by using a Handles clause on it). This will be a breaking change for a consumer of a converted VB library. If people complain about this scenario, we could add an option to toggle this trade-off since it seems like it will benefit a large proportion of users.

### Manual test case tried

Real world app with Form2 inheriting Form1 and adding a method to handle a button click:
* Check events fire and, and that the designer renders
* After conversion, make a change in the designer and save
* Check it's still the case that events fire and the designer renders.
